### ISSUE TITLE
The implementation of the peer layer seems to be wrong

### ISSUE THREAD

**Issue by autumn-2-net** [huyphan168/PEER#3]
https://github.com/huyphan168/PEER/blob/9d8ea04821c5aa4987733e105f3dbd596a5e89ca/peer/model.py#L94
The correct version should like this
(scores_x,indices_x), ( scores_y, indices_y) = [s.topk(self.product_key_topk, dim=-1) for s in sim]

**Comment by ShifengXIE** [comment]
I agree with the proposal. I tested this code on a few vision tasks. Better results will be obtained after this modification.

**Comment by evan-py-chen** [comment]
Another issue:
https://github.com/huyphan168/PEER/blob/9d8ea04821c5aa4987733e105f3dbd596a5e89ca/peer/model.py#L109C1-L110C54

It should be:
weights_down = self.weight_down_embed(indices)
 weights_up = self.weight_up_embed(indices)
